configuration/pih/scripts/global/ces.js

```diff
@@ -274,82 +274,107 @@ function setUpPlanSection(
-    medP.css("display", "inline");
+async function printPrescription(formattedConsultDate, patientName, age, diagnoses, prescriptions) {
+
+  // Load in the PDF template
+  const templateUrl = emr.resourceLink('file', 'configuration/pih/images/recetas-template.pdf');
+  const existingPdf = await fetch(templateUrl).then(res => res.arrayBuffer());
+  const pdfDoc = await PDFLib.PDFDocument.load(existingPdf);
+
+  // Add prescription data
+  const form = pdfDoc.getForm();
+  form.getTextField('date').setText(formattedConsultDate);
+  form.getTextField('patientName').setText(patientName);
+  form.getTextField('age').setText('' + age);
+  form.getTextField('diagnosis').setText(diagnoses);
+
+  const pageFont = await pdfDoc.embedFont(PDFLib.StandardFonts.Helvetica);
+  const pageFontSize = 10;
+  const page = pdfDoc.getPages()[0];
+
+  let rowPosition = 450;
+
+  page.drawText('Nombre del medicamento', {x: 30, y: rowPosition, size: pageFontSize, font: pageFont});
+  page.drawText('Frascos/Cajas', {x: 350, y: rowPosition, size: pageFontSize, font: pageFont});
+  page.drawText('Instrucciones', {x: 450, y: rowPosition, size: pageFontSize, font: pageFont});
+
+  page.drawLine({
+    start: { x: 30, y: 445 },
+    end: { x: 750, y: 445 },
+    thickness: 1,
+    color: PDFLib.rgb(1.0,0.59608, 0.0),
+  })
+
+  prescriptions.forEach(p => {
+    let drugNameLines = splitToLines(p.drugName ?? '', 65);
+    let instructionLines = splitToLines(p.instructions ?? '', 80);
+    let maxlines = drugNameLines.length > instructionLines.length ? drugNameLines.length : instructionLines.length;
+    for (var i=0; i<maxlines; i++) {
+      rowPosition = rowPosition - 20;
+      if (drugNameLines.length > i) {
+        page.drawText(drugNameLines[i], {x: 30, y: rowPosition, size: pageFontSize, font: pageFont});
+      }
+      if (i === 0) {
+        page.drawText(p.amount ?? '', {x: 350, y: rowPosition, size: pageFontSize, font: pageFont});
+      }
+      if (instructionLines.length > i) {
+        page.drawText(instructionLines[i] ?? '', {x: 450, y: rowPosition, size: pageFontSize, font: pageFont});
+      }
+    }
+    page.drawLine({
+      start: { x: 30, y: rowPosition-5 },
+      end: { x: 750, y: rowPosition-5 },
+      thickness: 1,
+      color: PDFLib.rgb(1.0,0.59608, 0.0),
+      opacity: 0.33,
+    })
+  });
+
+  /**
+   * TODO: This does not currently handle situations where the prescription content exceeds the space on the PDF
+   * This code supports 16-17 lines of prescription data.  Most medication orders are expected to take up one line
+   * Most prescriptions have maximum 3-4 medications.
+   * The drug with the longest name takes up 3 lines.  Instructions can take up arbitrary numbers of lines.
+   * If the need arises to support prescriptions with more content that is allowed, we can adapt this code to
+   * clone the first page of the PDF and add as many pages as needed given the expected number of lines of content
+   */
```

Comment:
Sounds fine to me, as long we feel it will generally be clear when it fails?  ie I'd rather it print all over the footer and be a mess rather than truncating so that it fits on one page and the user doesn't realize that one or more of the drugs are missing... hope that makse sense?

Reply:
Yes, it will be clear when it fails because the drugs will run over the footer and look a mess and it should be obvious that there was more data than could fit.  It's possible it'll throw an error if it goes outside of the allowable print area.  Not sure if we care to deal with that.